src/pages/index.md

```diff
@@ -2,8 +2,43 @@
 tags: ["intro"]
 ---
 
-CosmWasm is a smart contract platform focusing on security, performance, and
-interoperability. It is the only smart contracting platform for public
-blockchains with heavy adoption outside of the EVM world.
+The [Cosmos Wasm](https://github.com/CosmWasm/wasmd/tree/main/x/wasm), the root of CosmWasm, is a
+[CosmosSDK](https://docs.cosmos.network/) module enabling Smart Contracts to execute on the Wasm virtual machine.
+It is a bridge between Cosmos chain and the [WasmWM](https://github.com/CosmWasm/wasmvm) executing the Smart Contracts.
+CosmWasm itself is the whole ecosystem built around it with a missing to make Smart Contracts development easy and reliable.
+The focuses of the CosmWasm platform are security, performance, and interoperability. CosmWasm is the only Smart Contracts
+platform for public blockchains that has been heavily adopted outside the EVM world.
 
-Here you can find its documentation.
+We chose to target a Rust programming language as a Smart Contract development stack, as it has the best Wasm compiler on the
+market so far. As of today, we do not provide bindings to help write Smart Contracts in another stack that compiles to Wasm,
+and we don't support that.
+
+Here is where to find CosmWasm in the whole Cosmos stack:
+
+```mermaid
+erDiagram
+    CosmosSDK ||--|| BFT: Uses
+    CosmosSDK ||--|| "Wasm/WasmWM": Includes
+    CosmosSDK ||--o{ "Custom Module" : Includes
+    "Wasm/WasmWM" ||--o{ "Smart Contract": Executes
+
+```
+
+The important thing about CosmWasm Smart Contracts is their transparency. Even though they are executed by the chain node application
+using the WasmVM module, they are written to hide their dependency on it. Every Smart Contract has its unique address on the chain,
+and it can act just like any other chain client. It is easy to implement communication between two Smart Contracts on the same chain.
```

Comment:
```suggestion
and it can act just like any other chain client. It is easy to implement communication between two smart contracts on the same chain.
```